### DISCUSSION TITLE
While Copying Text From Apps Like Figma in Project Data it adds a attribute with long string

### DISCUSSION THREAD

**Discussion by RakulAgn** [GrapesJS/grapesjs#5682] [General]
When copying text from Figma and pasting it into a text block, it adds attributes with long text, which may lead to database errors when storing the GrapesJS project data. Is this usual behavior, or is it an issue? Is there any way to prevent this kind of issue?

` "components": [
                      {
                        "type": "text",
                        "attributes": {
                          "id": "ij3f"
                        },
                        "components": [
                          {
                            "components": [
                              {
                                "tagName": "span",
                                "attributes": {
                                  "data-metadata": ""
                                }
                              },
                              {
                                "tagName": "span",
                                "attributes": {
                                  "data-buffer": ""
                                }
                              }
                            ]
                          },
                          {
                            "tagName": "span",
                            "type": "text",
                            "attributes": {
                              "id": "i4il"
                            },
                            "components": [
                              {
                                "type": "textnode",
                                "content": "Real-Time Actionable Inputs To Grow Sales"
                              }
                            ]
                          }
                        ]
                      }
                    ]`

**Comment by artf** [comment]
>  Is this usual behavior, or is it an issue?

Kind of expected, looks like Figma copies an HTML type data (probably for internal purposes) + plain text. When you paste from the clipboard, the HTML type is used if the target element supports it (eg. contentEditable elements), plain text otherwise (eg. input elements).

> Is there any way to prevent this kind of issue?

Yes, but currently we have our own [logic](https://github.com/GrapesJS/grapesjs/blob/b6613061eb63ce7acbf513dbb84625b0c17b6a33/src/rich_text_editor/model/RichTextEditor.ts#L308) to handle paste event. 
I can add, for the next release, new options to allow overwriting the built-in logic. For now, you should be able to monkey-patch the `editor.RichTextEditor.globalRte.__onPaste` method with your own logic.
